### ISSUE TITLE
Edit/"Select All" Menu item disabled with single project selected in Project Explorer

### ISSUE THREAD

**Issue by dnapoletanoexp** [eclipse-jdt/eclipse.jdt.core#2988]
The "Select All" action in Project Explorer does not work when only a project is clicked; it works only after clicking another.
Steps for reproducing the problem:
- Open Eclipse on a workspace with two or more projects
- In Project Explorer, click on a project
- Click on "Edit" menu: the "Select All" item is disabled (it should be enabled), and CTRL+A has no effect (it should select all projects)
- In Project Explorer, click on another project (or click again on the same project)
- Click again on the "Edit" menu: the "Select All" item is enabled, and CTRL+A works as expected

**Comment by AddRain1** [comment]
I'm working on this

**Comment by HannesWell** [comment]
Great. You are one of the students from CodeDays, aren't you?

Much success on this task and don't hesitate to ask for help if you need any.

**Comment by AddRain1** [comment]
> Great. You are one of the students from CodeDays, aren't you?
> 
> Much success on this task and don't hesitate to ask for help if you need any.

Yes, thank you very much.

**Comment by Xiaotian-Lyu** [comment]
Hi, I am also working on this issue. I am in the same team with adrian. 
I installed the latest version and cannot repro this issue.

Some operations I have tried so far:
- Open Eclipse workspace and directly click "Select All"
- Open Eclipse workspace, click on one project and then click "Select All"
- Open Eclipse workspace and "Ctri+A"
- Open Eclipse workspace, click on one project and then click "Ctri+A"

It seems to work fine. 
This is the version I tried:
Version: 2024-12 (4.34)
Build id: I20241014-1810

I am wondering if this issue only exists on a specific version?

**Comment by dnapoletanoexp** [comment]
> Hi, I am also working on this issue. I am in the same team with adrian. I installed the latest version and cannot repro this issue.
> 
> Some operations I have tried so far:
> 
> * Open Eclipse workspace and directly click "Select All"
> * Open Eclipse workspace, click on one project and then click "Select All"
> * Open Eclipse workspace and "Ctrl+A"
> * Open Eclipse workspace, click on one project and then click "Ctrl+A"
> 
> It seems to work fine. This is the version I tried: Version: 2024-12 (4.34) Build id: I20241014-1810 Java 23
> 
> I am wondering if this issue only exists on a specific version? Also, cc @AddRain1

Version I'm using is 2024-09, the latest available from https://eclipseide.org/ . Perhaps the problem has disappeared in upcoming 2024-12 as a side effect of some other development?

**Comment by AddRain1** [comment]
@dnapoletanoexp @Xiaotian-Lyu 
I believe the issue has already been resolved in the upcoming 2024-12 version. My teammate and I were able to reproduce the issue on 2024-09, but were unable to in 2024-12. I wasn't able to find any duplicate issues. @dnapoletanoexp Would you mind looking at the 2024-12 version?

**Comment by dnapoletanoexp** [comment]
> @dnapoletanoexp @Xiaotian-Lyu I believe the issue has already been resolved in the upcoming 2024-12 version. My teammate and I were able to reproduce the issue on 2024-09, but were unable to in 2024-12. I wasn't able to find any duplicate issues. @dnapoletanoexp Would you mind looking at the 2024-12 version?

I've tried with 2024-12 M2, "Eclipse IDE for Java Developers" (Windows x86_64) downloaded from here:

https://www.eclipse.org/downloads/packages/release/2024-12/m2

creating a new workspace and 3 Java projects, but the problem persists. Further specs:
- OS: Windows 10 x64
- Java:  "21" 2023-09-19 LTS

**Comment by dnapoletanoexp** [comment]
Update: I've tried with 2024-12 M3, Eclipse IDE for Java Developers:

[https://www.eclipse.org/downloads/packages/release/2024-12/m3](https://www.eclipse.org/downloads/packages/release/2024-12/m3)

Selection works well in **Package** Explorer; the problem still occurs in **Project** explorer.

**Comment by dnapoletanoexp** [comment]
Further update: I've downloaded the "Eclipse IDE for Enterprise Java and Web Developers" release version from:

[https://www.eclipse.org/downloads/packages/](https://www.eclipse.org/downloads/packages/)

The problem still occurs.

**Comment by srikanth-sankaran** [comment]
@jarthana - should this be forwarded to platform UI ?? Or JDT UI ??

**Comment by jarthana** [comment]
I believe this belongs in Platform. But I don't think we can transfer issues outside JDT. We will have to create a new one.
@HannesWell Do you know which repo that would be?

**Comment by HannesWell** [comment]
> I believe this belongs in Platform. But I don't think we can transfer issues outside JDT. We will have to create a new one. 

Unfortunately yes. IIRC the EF-infra team is working on some bots to help with that, but that's not yet available.

> @HannesWell Do you know which repo that would be?

From the description I think it's https://github.com/eclipse-platform/eclipse.platform.ui .

**Comment by dnapoletanoexp** [comment]
Created as new issue here:

[https://github.com/eclipse-platform/eclipse.platform.ui/issues/2739](https://github.com/eclipse-platform/eclipse.platform.ui/issues/2739)
